### DISCUSSION TITLE
Share Firestore Cache with Anglar PWA

### DISCUSSION THREAD

**Discussion by laborc8** [TruCopilot/phpfastcache#845] [General]
The idea is to WRITE and UPDATE the cache via the php System and then READ via Angular PWA (or Ionic Hyprid-App).
Both access the firestore database.
As I am in the process of getting it up and running (for a proof-of-concept) I have not seen the firestore structure yet.
So my question would be: Do you think this is possible?
How can I access the objects (also with tags) from the firestore db.
Can I do it by basic firestore api calls? Are they stored by key only or is there a specifc class in php that handles that access?
Thanks for you time and ideas.
all the best Mario

**Comment by Geolim4** [comment]
Hello,

The Firestore access is done via the PSR-6 interfaces handled by Phpfastcache.

So Phpfastcache only needs GCP environment variables to be set and the config object filled with Google project and firestore database name:
https://github.com/PHPSocialNetwork/phpfastcache/blob/bcd068ff1a78cd1475a4d8c13ed371711465ef40/lib/Phpfastcache/Drivers/Firestore/Config.php#L1
